### PR TITLE
Activate ruff print-t20

### LINKED ISSUE #8997: Enable more rules in ruff
We recently removed to ruff (https://github.com/mlflow/mlflow/pull/8873) to lint python files faster. We can enable more rules. This is a ticker for tracking that. To make PRs reviewable, we should enable rules one by one.

Rules that I think are useful:

- [x] https://beta.ruff.rs/docs/rules/redundant-open-modes/ (@EdAbati)
- [ ] https://beta.ruff.rs/docs/rules/#flake8-print-t20 (@Bncer)

Example PR:

https://github.com/mlflow/mlflow/pull/8958

### PR DESCRIPTION
**Pull request by bncer**
## Related Issues/PRs

Resolve #8997

## What changes are proposed in this pull request?

As stated in [comment](https://github.com/mlflow/mlflow/issues/8997#issuecomment-1627591495) this PR activates ruff [T20](https://beta.ruff.rs/docs/rules/#flake8-print-t20) rule for print statements and applied to project. Necessary print statements saved by `per-file-ignores`

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -70,3 +72,7 @@ exclude = [
   "mlflow/temporary_db_migrations_for_pre_1_users",
   "tests/protos",
 ]
+[tool.ruff.per-file-ignores]
+"dev/*" = ["T201", "T203"]
+"tests/resources/example_project/greeter.py" = ["T201", "T203"]
```
Comment (harupy) [inline pyproject.toml:77]: For this file, use `noqa: T201`

File: pyproject.toml [hunk 2]
```diff
@@ -70,3 +72,7 @@ exclude = [
...
+"tests/resources/mlflow-test-plugin/mlflow_test_plugin/fake_deployment_plugin.py" = ["T201", "T203"]
```
Comment (harupy) [inline pyproject.toml:78]: https://github.com/mlflow/mlflow/pull/9010/files#r1257466602 same comment on this file

File: pyproject.toml [hunk 3]
```diff
@@ -70,3 +72,7 @@ exclude = [
   "mlflow/temporary_db_migrations_for_pre_1_users",
   "tests/protos",
 ]
+[tool.ruff.per-file-ignores]
+"dev/*" = ["T201", "T203"]
```
Comment (harupy) [inline pyproject.toml:76]: do we need T203?

Comment (Bncer) [reply pyproject.toml:76]: No, I will delete it. Thought to add because task was to add T20 )

**Comment by harupy** [comment]
@Bncer can you address https://github.com/mlflow/mlflow/issues/8997#issuecomment-1627591495?

File: pyproject.toml (hunk 1 above)
Comment (Bncer) [reply pyproject.toml:77]: Ok

**Comment by Bncer** [comment]
> @Bncer can you address [#8997 (comment)](https://github.com/mlflow/mlflow/issues/8997#issuecomment-1627591495)?

Oh yeah! Will do it

**Comment by Bncer** [comment]
@harupy Do I need to close PR and recreate from local branch or can add next commit to fix?

**Comment by harupy** [comment]
You can add new commits :)

File: pyproject.toml [hunk 4]
```diff
@@ -25,6 +25,8 @@ select = [
   "UP015",
   "UP031",
   "UP034",
+  "T201",
+  "T203",
```
Comment (harupy) [inline pyproject.toml:29]:
```suggestion
  "T20",
```

File: tests/resources/mlflow-test-plugin/mlflow_test_plugin/fake_deployment_plugin.py [hunk 5]
```diff
@@ -53,7 +53,7 @@ def get_endpoint(self, endpoint):
 
 def run_local(name, model_uri, flavor=None, config=None):
     # pylint: disable-next=print-function
```
Comment (harupy) [inline tests/resources/mlflow-test-plugin/mlflow_test_plugin/fake_deployment_plugin.py:55]:
```suggestion
```

File: tests/resources/example_project/greeter.py [hunk 6]
```diff
@@ -14,4 +14,4 @@
     if args.excitement is not None:
         greeting.append("!" * args.excitement)
     # pylint: disable-next=print-function
```
Comment (harupy) [inline tests/resources/example_project/greeter.py:16]:
```suggestion
```

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**